### PR TITLE
Add cache buster to webpack chunks

### PR DESCRIPTION
**Pull request by juliusknorr**
This should fix the browser caching issues after Nextcloud updates. We now append the chunk hash when requesting other files than the entry points. Entry points get the version cachebuster from nextcloud itself.

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]
:+1: Makes sense

I ran into this for Mail as well https://github.com/nextcloud/mail/commit/24ccca2cd46d5dd05259b1766f91566fbdbe3453

File: apps/files_sharing/webpack.js [hunk 1]
```diff
@@ -10,6 +10,6 @@ module.exports = {
 		path: path.resolve(__dirname, './js/dist/'),
 		publicPath: '/js/',
 		filename: '[name].js',
-		chunkFilename: 'files_sharing.[id].js'
+		chunkFilename: 'files_sharing.[id].js?v=[chunkhash]'
```
Comment (skjnldsv) [inline apps/files_sharing/webpack.js]: Does it work like that?
Shouldn't we need to have this in the php add_script function?

Comment (ChristophWurst) [reply apps/files_sharing/webpack.js]: No, this changes the way URLs for asynchronously loaded chunks looks like. PHP will never see those.

Comment (juliushaertl) [reply apps/files_sharing/webpack.js]: The php add_script does it for the entrypoint but for dynamic imports this PR will fix it ;)

Comment (skjnldsv) [reply apps/files_sharing/webpack.js]: Ah yes, chunks are loaded from js, I forgot :)

**Review by rullzer — APPROVED** [review]

**Pull request merged**